tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -242,9 +242,7 @@ void EnqueueWriteShardedBufferCommand::add_dispatch_write(HugepageDeviceCommand&
     uint32_t data_size_bytes = this->pages_to_write * this->padded_page_size;
     const CoreCoord virtual_core =
         this->buffer.device()->virtual_core_from_logical_core(this->core, this->buffer.core_type());
-    bool flush_prefetch = true;
```

Comment:
`EnqueueWriteBufferCommand::process()` sizes the total cmd size as
```
uint32_t cmd_sequence_sizeB =
        CQ_PREFETCH_CMD_BARE_MIN_SIZE +  // CQ_PREFETCH_CMD_RELAY_INLINE + (CQ_DISPATCH_CMD_WRITE_PAGED or
                                         // CQ_DISPATCH_CMD_WRITE_LINEAR)
        data_size_bytes;
```

Should it be more like this?
```
uint32_t cmd_sequence_sizeB = align(
        sizeof(CQPrefetchCmd) +
        sizeof(CQDispatchCmd) + 
        data_size_bytes, pcie_alignment);
```

Reply:
hm I thought this would be okay because prefetch cmd relay inline will make the stride pcie aligned so it would be `align(sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd), pcie alignment)` which should be `CQ_PREFETCH_CMD_BARE_MIN_SIZE`

Reply:
Right, that would be what the device strides for the next command, but this part is for host doing issue queue reserve/push. `CQ_PREFETCH_CMD_BARE_MIN_SIZE + data_size_bytes` wouldn't necessarily be equal to what we calculate as stride? Stride calculation would be `align(sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd) + data_size_bytes, pcie alignment)`

Reply:
good point, Ill update this

Reply:
We really need to expose methods of calculating sizes for cmds to simplify this, since right now we basically manually recalculate in command_queue.cpp what device_command.hpp is doing which is really error prone 😞

Reply:
yeah I can open up an [issue](https://github.com/tenstorrent/tt-metal/issues/16149) for this but I think I won't address that holistically in this PR

Reply:
Yeah that makes sense. Sorry should have clarified I wasn't expecting that to be done in this pr